### ISSUE TITLE
Make comments in secrets possible

### ISSUE THREAD

**Issue by jugmac00** [flyingcircusio/batou#77]
I guess we do not have to discuss about the general value of comments (hello JSON!).

But there is also one very specific need.

When saving `htpasswd` data as string in a secret, I'd like to add the plain text password to avoid hassle and the drift of information when saving the password encrypted in the secret and plain text at another place.

Currently, when I try to add a comment....

```
[component:nginx]
credentials = staging:someHashString
# plain text password: does not work
```

I receive an error message, both when I try to add a comment via `#` and `;`.

```
Could not update due to error: GPG returned non-zero exit code.
Your changes are still available. You can try:
        edit (opens editor with current data again)
        encrypt (tries to encrypt current data again)
        quit (quits and loses your changes)
```

The current workaround - also known as let's jump on the JSON bandwaggon - is to declare a second key/value pair, which then should be ignored by the dev/batou, ie

```
[component:nginx]
credentials = staging:someHashString
plain_text_password = does-work-but-is-kinda-confusing
```

Thanks! :-)

**Comment by ctheune** [comment]
A note on the use case. I think we _should_ support comments and will look into it. For your specific use case, I'd suggest you store the plain text password in the secrets and use python functions to compute the hash ... ;)

**Comment by ctheune** [comment]
I'm a bit surprised about the error you are getting - in my cases (either using ; or #) I'm just loosing my data (because the parse/serialize cycle "eats them") but not getting an error.

**Comment by jugmac00** [comment]
> I'm a bit surprised about the error you are getting - in my cases (either using ; or #) I'm just loosing my data (because the parse/serialize cycle "eats them") but not getting an error.

I also recall just loosing the data from quite some time ago... but well, I just tried it again. I get above error message.

This is on Ubuntu 18.04 with gpg 2.2.4.

The loosing data behavior I recall may date back to the time I used Windows. (maybe)

And thanks for the hint about saving plain text and calculating the hash in Python - sounds good!

**Comment by ctheune** [comment]
Fixed in b252f7a0b4f8d12cf5ea2e70c3fc3dad0ebaa516
